python-frontend/src/main/java/org/sonar/python/semantic/v2/SymbolsModuleTypeProvider.java

```diff
@@ -99,26 +99,48 @@ private static PythonType convertToObjectType(Symbol symbol) {
     return PythonType.UNKNOWN;
   }
 
-  private static PythonType convertToFunctionType(FunctionSymbol symbol) {
-    return new FunctionType(symbol.name(), List.of(), List.of(), PythonType.UNKNOWN, false, false, false, false, null);
+  private static PythonType convertToFunctionType(FunctionSymbol symbol, Map<Symbol, TypeBuilder> createdTypesBySymbol) {
```

Comment:
I think it will be easier to operate with PythonTypes instead of TypeBuilders map here